x-pack/legacy/plugins/remote_clusters/public/app/sections/components/remote_cluster_form/remote_cluster_form.js

```diff
@@ -516,57 +518,69 @@ export class RemoteClusterForm extends Component {
       return null;
     }
 
-    const errorExplanation = [];
+    const errorExplanations = [];
 
     if (errorClusterName) {
-      errorExplanation.push({
+      errorExplanations.push({
         key: 'nameExplanation',
-        where: i18n.translate('xpack.remoteClusters.remoteClusterForm.inputNameErrorsScreenReaderExplanation', {
-          defaultMessage: 'The input "Name". You have entered: {name}.',
-          values: { name }
+        field: i18n.translate('xpack.remoteClusters.remoteClusterForm.inputNameErrorMessage', {
+          defaultMessage: 'The "Name" field is invalid.',
+        }),
+        error: errorClusterName
+      });
+    }
+
+    if (errorsSeeds) {
+      errorExplanations.push({
+        key: 'seedsExplanation',
+        field: i18n.translate('xpack.remoteClusters.remoteClusterForm.inputSeedsErrorMessage', {
+          defaultMessage: 'The "Seed nodes" field is invalid.',
         }),
-        what: errorClusterName
+        error: errorsSeeds
       });
     }
+
     if (localSeedErrors && localSeedErrors.length) {
-      errorExplanation.push({
-        key: 'seedExplanation',
-        where: i18n.translate('xpack.remoteClusters.remoteClusterForm.inputSeedErrorsScreenReaderExplanation', {
-          defaultMessage: 'The "Seed Nodes" input. You have entered: {seedInput}.',
-          values: { seedInput }
+      errorExplanations.push({
+        key: 'localSeedExplanation',
+        field: i18n.translate('xpack.remoteClusters.remoteClusterForm.inputLocalSeedErrorMessage', {
+          defaultMessage: 'The "Seed nodes" field is invalid.',
         }),
-        what: localSeedErrors.join(' ')
+        error: localSeedErrors.join(' '),
       });
     }
 
-    const messagesToBeRendered = (
-      errorExplanation.length && errorExplanation.map(errorMessage => (
-        <EuiScreenReaderOnly key={errorMessage.key}>
-          <p>
-            {errorMessage.where}<br/>
-            {errorMessage.what}
-          </p>
-        </EuiScreenReaderOnly>
-      ))
+    const messagesToBeRendered = errorExplanations.length && (
+      <EuiScreenReaderOnly>
+        <dl id={this.state.generateId(ERROR_LIST_ID)} aria-labelledby={this.state.generateId(ERROR_TITLE_ID)}>
```

Comment:
When the user hits "Save", these errors are read aloud, which is great, but the callout message is _not_ read aloud. Should we repeat "Fix errors before continuing" from line 574 within this `<EuiScreenReaderOnly>` element to give the user some context?

Reply:
I added the heading id to the described by attribute so now it reads of both.

The whole thing is a little duplicative because the fields announce their own errors and then the button announces the same errors again but with slightly more info...

It probably isn't the most ideal thing in the world but I think it's a good improvement over what we have today.